From: Scott Howard <showard@debian.org>
Date: Wed, 28 Mar 2018 15:52:22 -0300
Subject: no_macosx_think_different

---
 app/src/processing/app/Base.java | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/app/src/processing/app/Base.java b/app/src/processing/app/Base.java
index 477f931..f64235f 100644
--- a/app/src/processing/app/Base.java
+++ b/app/src/processing/app/Base.java
@@ -49,7 +49,6 @@ import processing.app.helpers.filefilters.OnlyDirs;
 import processing.app.helpers.filefilters.OnlyFilesWithExtension;
 import processing.app.javax.swing.filechooser.FileNameExtensionFilter;
 import processing.app.legacy.PApplet;
-import processing.app.macosx.ThinkDifferent;
 import processing.app.packages.LibraryList;
 import processing.app.packages.UserLibrary;
 import processing.app.syntax.PdeKeywords;
@@ -132,14 +131,6 @@ public class Base {
     }
     System.setProperty("java.net.useSystemProxies", "true");
 
-    if (OSUtils.isMacOS()) {
-      System.setProperty("apple.laf.useScreenMenuBar",
-        String.valueOf(!System.getProperty("os.version").startsWith("10.13")
-          || com.apple.eawt.Application.getApplication().isAboutMenuItemPresent()));
-
-      ThinkDifferent.init();
-    }
-
     try {
       INSTANCE = new Base(args);
     } catch (Throwable e) {
