From: Scott Howard <showard@debian.org>
Date: Wed, 28 Mar 2018 15:52:21 -0300
Subject: wrapper_for_deb

---
 build/linux/dist/arduino | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/linux/dist/arduino b/build/linux/dist/arduino
index ad97c91..3f94265 100755
--- a/build/linux/dist/arduino
+++ b/build/linux/dist/arduino
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-APPDIR="$(dirname -- "$(readlink -f -- "${0}")" )"
+APPDIR=/usr/share/arduino
 
 for LIB in \
     "$APPDIR"/java/lib/rt.jar \
@@ -12,7 +12,7 @@ do
 done
 export CLASSPATH
 
-LD_LIBRARY_PATH=$APPDIR/lib${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
+LD_LIBRARY_PATH="/usr/lib/jni${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
 export LD_LIBRARY_PATH
 
 export PATH="${APPDIR}/java/bin:${PATH}"
